File: frontend/src/app/shared/models/Claim.ts
import { ClaimStatus } from '../enums/ClaimStatus';
import { Client } from './Client';
import { InsuranceContract } from './InsuranceContract';
import { Compensation } from './Compensation';
import { Document } from './Document';
import { AutoClaimDetails } from './AutoClaimDetails';
import { HealthClaimDetails } from './HealthClaimDetails';
import { HomeClaimDetails } from './HomeClaimDetails';

export interface Claim {
    claimId: number;
    claimDate: Date;
    claimedAmount: number;
    approvedAmount: number;
    description: string;
    status: ClaimStatus;
    fraud: boolean;
    message?: string;
    
    // Relations
    client?: Client;
    contract?: InsuranceContract;
    compensation?: Compensation;
    documents?: Document[];
    autoDetails?: AutoClaimDetails;
    healthDetails?: HealthClaimDetails;
    homeDetails?: HomeClaimDetails;
}

File: backend/src/main/java/org/example/projet_pi/entity/Claim.java
package org.example.projet_pi.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;  // ← IMPORT
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

@Getter
@Setter
@Entity
public class Claim {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long claimId;

    @Temporal(TemporalType.DATE)
    private Date claimDate;

    private Double claimedAmount;
    private Double approvedAmount;

    private String description;

    @Enumerated(EnumType.STRING)
    private ClaimStatus status;

    @ManyToOne
    @JsonIgnore  // ← AJOUTER CEÇI
    private Client client;

    @ManyToOne
    @JsonIgnore  // ← AJOUTER CEÇI
    private InsuranceContract contract;

    @OneToOne(mappedBy = "claim", cascade = CascadeType.ALL)
    @JsonIgnore  // ← AJOUTER CEÇI
    private Compensation compensation;

    @OneToMany(mappedBy = "claim", cascade = CascadeType.ALL)
    @JsonIgnore  // ← AJOUTER CEÇI
    private List<Document> documents= new ArrayList<>();


    private Boolean fraud = false;
    private String message;

    @OneToOne(mappedBy = "claim", cascade = CascadeType.ALL)
    @JsonIgnore  // ← AJOUTER CEÇI
    private AutoClaimDetails autoDetails;

    @OneToOne(mappedBy = "claim", cascade = CascadeType.ALL)
    @JsonIgnore  // ← AJOUTER CEÇI
    private HealthClaimDetails healthDetails;

    @OneToOne(mappedBy = "claim", cascade = CascadeType.ALL)
    @JsonIgnore  // ← AJOUTER CEÇI
    private HomeClaimDetails homeDetails;

}
